tdwii_plus_examples/basescu.py

```diff
@@ -103,73 +111,126 @@ def __init__(
 
     def _add_requested_context(self):
         """
-        Adds the DICOM UPS Push SOP Class presentation context to the AE.
+        Adds the DICOM Verification SOP Class presentation context to the AE.
         Default transfer syntaxes are included.
 
-        This method may be overridden in derived classes if another SOP
-        Class needs to be negotiated.
+        Derived classes *must* override this method to add the presentation
+        contexts for the specific SOP Classes they support.
         """
         self.ae.add_requested_context(Verification)
         self.logger.debug(f"Verification Presentation context added: \n {self.ae.requested_contexts[0]}")
 
     # Create a named tuple to hold the result of the Association
     AssociationResult = namedtuple("AssociationResult", ["status", "description", "accepted_sop_classes"])
 
-    def _associate(self):
+    def _associate(self, required_sop_classes=None, verbose: bool = True):
         """
-        Establish an association with the remote AE and check contexts.
+        Establishes an association with the remote AE.
+
+        Returns an AssociationResult named tuple containing the status, accepted SOP classes, and a description.
+        The status is "Success" if all requested presentation contexts are accepted, "Warning" if only some are
+        accepted and "Error" if the association fails.
 
         Returns:
-            AssociationResult: A named tuple containing the status, accepted SOP classes, and description.
+            An AssociationResult namedtuple with the following fields:
+
+            - status (str): The status of the association ("Success", "Warning", or "Error").
+            - description (str): A description of the association result.
+            - accepted_sop_classes (list of UID): A list of accepted SOP Class UIDs.
         """
+
+        result = self._attempt_association(required_sop_classes)
+        return result if verbose else (result.status == "Success")
```

Comment:
What about returning a tuple with the first value being the boolean and the second being the named tuple.
That would make it consistent, and current callers that are specifying verbose=False can just do something like:

success, _ = my_scu._associate(...)

while those who want the more complete version

success, details = my_scu._associate(...)

Reply:
@dwikler just want to draw your attention to the above success, details = my_scu._associate(...) suggestion.
I think pynetdicom uses an approach similar to this in some scenarios.